src/files/infra/repositories/FilesRepository.ts

```diff
@@ -202,6 +206,52 @@ export class FilesRepository extends ApiRepository implements IFilesRepository {
       })
   }
 
+  public async getFileUploadDestination(
+    datasetId: number | string,
+    file: File
+  ): Promise<FileUploadDestination> {
+    return this.doGet(
+      this.buildApiEndpoint(this.datasetsResourceName, 'uploadurls', datasetId),
+      true,
+      {
+        size: file.size
+      }
+    )
+      .then((response) => transformUploadDestinationsResponseToUploadDestination(response))
+      .catch((error) => {
+        throw error
+      })
+  }
+
+  public async addUploadedFileToDataset(
+    datasetId: number | string,
+    uploadedFileDTO: UploadedFileDTO
+  ): Promise<undefined> {
```

Comment:
Could be helpful for the consumer of the use case to receive something different that undefined here if the operation suceeds ? what will be the final output that the use case will send?

Reply:
This method is opaque to the use case (In fact, the use case calling this method returns Promise<void>). I see no value in changing the return type. In repository calls whose response do not return anything, we always return undefined.